### PR TITLE
Fixed TorpedoStore.java

### PR DESCRIPTION
**Pull request by iet2021**
3 commits in TorpedoStore.java (complaints by SonarLit)

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -14,7 +14,10 @@
 
   private int torpedoCount = 0;
 
+  private Random generator;
```
Comment (stesoma) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: Random creation is now in the correct place

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +33,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (stesoma) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Thrown, like it should be. Good, but exception message could be more descriptive

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +33,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (stesoma) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:46]: Correct. Was it a typo?

**Review by stesoma — CHANGES_REQUESTED** [review]
Changes OK.
Please make exception message more descriptive, like which values could cause the problem

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 2 above)
Comment (iet2021) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Im on it!

**Review by stesoma — APPROVED** [review]
Now perfect

**Pull request merged**